tests/test_cli.py

```diff
@@ -35,83 +35,118 @@ def make_fake(path, empty):
 
 
 def test_cli_okcompare():
+    """Shoult exit 0: rasters are equals."""
     runner = CliRunner()
-    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif', 'tests/fixtures/notblobby.tif', '--upsample', '8', '--compare-masked', '--downsample', '64'])
+    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif',
+                                 'tests/fixtures/notblobby.tif',
+                                 '--upsample', '8',
+                                 '--compare-masked', '--downsample', '64'])
     assert result.exit_code == 0
 
 
 def test_cli_okcompare_bad():
+    """Shoult exit 1: rasters are different."""
     runner = CliRunner()
-    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif', 'tests/fixtures/notblobby.tif', '--upsample', '8', '--downsample', '64'])
-    assert result.exit_code == -1
+    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif',
+                                 'tests/fixtures/notblobby.tif'])
+    assert result.exit_code == 1
+    assert result.output == 'Error: not ok - Band 1 has 4356 pixels that vary by more than 16\n'
 
 
 def test_cli_okcompare_bad_no_error():
+    """Shoult exit 0: raster are different but  `--no-error` is set to True."""
     runner = CliRunner()
-    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif', 'tests/fixtures/notblobby.tif', '--upsample', '8', '--downsample', '64', '--no-error'])
+    result = runner.invoke(cli, ['compare', 'tests/expected/blobby.tif',
+                                 'tests/fixtures/notblobby.tif', '--no-error'])
     assert result.exit_code == 0
-    assert result.output == 'not ok - Band 1 has 363 pixels that vary by more than 16\n'
+    assert result.output == 'not ok - Band 1 has 4356 pixels that vary by more than 16\n'
```

Comment:
The assertion diff is a tip off that the inputs changed here - looks like we're missing the `'--upsample', '8', '--downsample', '64'` parameters.

Reply:
Well moving to rasterio 1.0b1 (and maybe a higher gdal version) made this test fail (resampling ?) with '--upsample', '8', '--downsample', '64'. So because `'tests/expected/blobby.tif'` and `'tests/fixtures/notblobby.tif' should be different anyway I removed those other options

Reply:
So there's two things going on here:

First, the actual purpose of the test is to check the `--no-error` option, which your new version does just as well as the old version.

But I'm concerned that the same inputs are not returning the same outputs. We need to understand where/why it changed upstream, make sure the change was intentional and makes sense, and make sure there are no breaking changes downstream for applications that use raster-tester.

Reply:
agreed

Reply:
I'm not sure what is going on here -- this comparison is `OK` with an up / down of 2, which indicates that the output now is quite different than the original which had an up/down of 8 + 64.

Reply:
Not sure if I should merge then ? @perrygeo @dnomadb

Reply:
So the new tests look great and the intent is much more clear. The difference can be narrowed down to Rasterio version 


With Rasterio 0.36

```
⭆ raster-tester compare 'tests/expected/blobby.tif' 'tests/fixtures/notblobby.tif' --upsample 8 --downsample 64
ValueError: Band 1 has 363 pixels that vary by more than 16
```

With Rasterio 1.0b1

```
⭆ raster-tester compare 'tests/expected/blobby.tif' 'tests/fixtures/notblobby.tif' --upsample 8 --downsample 64
ok - tests/expected/blobby.tif is similar to within 0 pixels of tests/fixtures/notblobby.tif
```

So I think we should merge this, nothing we can do about the disrecpency on the raster-tester end. I'll follow up in Rasterio if needed.

Reply:
I'm 👍 on merging, especially if the output _looks_ OK @vincentsarago .

Reply:
merci